docker/build.mk

```diff
@@ -184,33 +184,44 @@ $(foreach fuzzer,$(FUZZERS), \
 
 
 define oss_fuzz_benchmark_template
-$(1)-project-name := $(shell cat benchmarks/$(1)/oss-fuzz.yaml | \
-                             grep project | cut -d ':' -f2 | tr -d ' ')
+# Use eval to make variables accessible within this template.
+$(eval $(1)-project-name := $(shell cat benchmarks/$(1)/oss-fuzz.yaml | \
+                             grep project | cut -d ':' -f2 | tr -d ' '))
 $(1)-fuzz-target  := $(shell cat benchmarks/$(1)/oss-fuzz.yaml | \
                              grep fuzz_target | cut -d ':' -f2 | tr -d ' ')
-$(1)-oss-fuzz-builder-hash := $(shell cat benchmarks/$(1)/oss-fuzz.yaml | \
+$(eval $(1)-oss-fuzz-builder-hash := $(shell cat benchmarks/$(1)/oss-fuzz.yaml | \
                                       grep oss_fuzz_builder_hash | \
-                                      cut -d ':' -f2 | tr -d ' ')
+                                      cut -d ':' -f2 | tr -d ' '))
+.$(1)-oss-fuzz-benchmark-wrapper-builder:
+	docker build \
+    --tag $(BASE_TAG)/builders/$(1)-oss-fuzz-benchmark-wrapper-builder \
+    --file=docker/oss-fuzz-benchmark-wrapper-builder/Dockerfile \
+    --build-arg parent_image=gcr.io/fuzzbench/oss-fuzz/$($(1)-project-name)@sha256:$($(1)-oss-fuzz-builder-hash) \
+    $(call cache_from,$(BASE_TAG)/builders/fuzz$($(1)-project-name)-builder) \
+    .
+
+.pull-$(1)-oss-fuzz-benchmark-wrapper-builder:
+	docker pull $(BASE_TAG)/builders/$(1)-oss-fuzz-benchmark-wrapper-builder
 endef
+
 # Instantiate the above template with all OSS-Fuzz benchmarks.
 $(foreach oss_fuzz_benchmark,$(OSS_FUZZ_BENCHMARKS), \
   $(eval $(call oss_fuzz_benchmark_template,$(oss_fuzz_benchmark))))
 
-
 define fuzzer_oss_fuzz_benchmark_template
 
-.$(1)-$(2)-oss-fuzz-builder-intermediate:
+.$(1)-$(2)-builder-intermediate: .$(2)-oss-fuzz-benchmark-wrapper-builder
 	docker build \
     --tag $(BASE_TAG)/builders/$(1)/$(2)-intermediate \
     --file=fuzzers/$(1)/builder.Dockerfile \
-    --build-arg parent_image=gcr.io/fuzzbench/oss-fuzz/$($(2)-project-name)@sha256:$($(2)-oss-fuzz-builder-hash) \
+    --build-arg parent_image=$(BASE_TAG)/builders/$(2)-oss-fuzz-benchmark-wrapper-builder \
     $(call cache_from,${BASE_TAG}/builders/$(1)/$(2)-intermediate) \
     fuzzers/$(1)
 
-.pull-$(1)-$(2)-oss-fuzz-builder-intermediate:
+.pull-$(1)-$(2)-builder-intermediate: .pull-$(2)-oss-fuzz-benchmark-wrapper-builder
 	docker pull $(BASE_TAG)/builders/$(1)/$(2)-intermediate
```

Comment:
Sorry, what do you mean? Why is this pull different from the other pulls (those are so that one can pull from gcr.io/fuzzbench to benefit from caching).

Reply:
Sorry what do you mean? Why is this pull unnecessary but others are? (The purpose of any pull is so local bulds can take advantage of caching based on what we built on GCB).

Reply:
Ignore, i think i was confused.